pilot/cmd/pilot-agent/main.go

```diff
@@ -278,11 +288,13 @@ var (
 				log.Infof("Effective config: %s", out)
 			}
 
+			sdsEnabled, sdsTokenPath := getSDSData()
```

Comment:
How about: sdsUdsPath, sdsTokenPath := detectSds() ?

Reply:
I mean: the point of this PR is to _detect_ that SDS exists on the node. 
Having an annotation doesn't help - the token will be there ( the unsecure one at least ).